### DISCUSSION TITLE
Tinyboard. Like Tensorboard, but tiny

### DISCUSSION THREAD

**Discussion by xnought** [tinygrad/tinygrad#4066] [Ideas]
https://www.tensorflow.org/tensorboard, but built specifically for tinygrad? tinyboard

A new dashboard with a similar (or exact same) API to wandb for saving and visualizing metrics, other debugging related stuff, or even visualizing the compiled model all done locally.

Thoughts? Maybe this could be a separate package?

**Comment by balloch** [comment]
Def a separate package, but I think it sounds like a great idea! In principle: tensorboard considers itself _associated_ with tensorflow, but in practice the core reason it is popular is because it is actually fully separable from tensorflow. Maybe more to the point: tinygrad is a library with a focused purpose: to free the ML compute of closed source. Tensorflow falls squarely outside of this because it doesn't run on GPU/acceleration so its not part of the compute problem. That said, I think that the community would greatly benefit if there was a non-corporation-associated alternative to tensorboard.
